### ISSUE TITLE
`Secure` flag for Cookies not working?

### ISSUE THREAD

**Issue by DavidGarciaCat** [symfony/symfony#26667]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | 2.8 & 3.x (pending), 4.x (confirmed)

I have been testing the new Symfony 4 with a micro-site, basically to have some content on a "dead" domain that I own. I chosed SF4 because it's the newest version, so the one that will be found everywhere in a near future (I guess).

Based on official [Symfony documentation](https://symfony.com/doc/current/reference/configuration/framework.html#cookie-secure), my settings for cookies should be adding the [`Secure` flag](https://github.com/DavidGarciaCat/davidgarcia.cat/blob/master/config/packages/framework.yaml#L11), however looks like this is not the case, and when I run a Web Inspection through tools like [pentest-tools.com](https://pentest-tools.com/website-vulnerability-scanning/web-server-scanner) or [observatory.mozilla.org](https://observatory.mozilla.org/) then I get "warnings" because the Cookies doesn't set this flag.

So I believe that there are 2 potential cases:
- I am missing an additional configuration that is needed (could be)
- There is a bug on SF settings that are not allowing me to set the flag

I didn't have time - yet - to test it with previous versions like SF 2.8 or SF 3.x, but these are some tests that I will do in the upcoming days for sure, so I will share my findings as soon as possible.

Any thoughts on this matter, and why is not setting the Secure flag?

Thanks in advance for your comments,

**Comment by xabbuh** [comment]
Could you create a small example application that allows to reproduce your issue?

**Comment by DavidGarciaCat** [comment]
Hi @xabbuh 

I guess that the same repo that I have now will work:
https://github.com/DavidGarciaCat/davidgarcia.cat

Where we can see the setting for `Secure` flag here:
https://github.com/DavidGarciaCat/davidgarcia.cat/blob/master/config/packages/framework.yaml#L11

There's a Docker config file, so you can test it locally

Inspection for this can be done through Observatory as pointed out on the issue's description:
https://observatory.mozilla.org/analyze/www.davidgarcia.cat

<img width="595" alt="screen shot 2018-03-29 at 16 21 05" src="https://user-images.githubusercontent.com/7809429/38097404-34ef5e2a-336d-11e8-8152-49ab68f2f90e.png">

Hope this helps,

**Comment by xabbuh** [comment]
Are you sure that's caused by the session cookie? I don't see where the session is started in your application. But there are other cookies (Google Analytics for example) that are sent by your application too.

**Comment by DavidGarciaCat** [comment]
Hello @xabbuh 

Regarding your point about Analytics, there are 2 conditionals that must return `true` to include them:
- tracking code must be provided through `.env` file; and
- environment must be `prod`

I have made sure that these 2 Env Vars are empty, and also added a set of new commits to the repository, to run more tests, and I think that I have found the problem, but it leads to a new question.

After add 2 new lines on my `ResponseListener` to return a Session Cookie and to add another non-Session Cookie, I can see that the Session Cookie is flagged as secure (as expected) but the 2nd Cookie is not flagged (that was unexpected), so I needed to add a 2nd `ResponseListener` to look for all cookies on my Response and then update the Secure flag for all of them, before to send the Response to the browser. Just once this last commit was deployed, the warning reported by those OnLine PenTest tools reported 👍

So, my concern is the need to add this process - manually - to make sure that all cookies have the Secure flag. Maybe I am wrong, but I was expecting to have all cookies flagged as Secure automatically, in the same way that the Session Cookie is set as `Secure`, with just 1 parameter in the configuration files.

What are your thoughts about this? Is just a misunderstanding on my end? Shouldn't be a process to do this automatically?
